VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitDiscussionFragment.java

```diff
@@ -0,0 +1,77 @@
+package org.edx.mobile.view;
+
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentTransaction;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import org.edx.mobile.R;
+import org.edx.mobile.model.api.EnrolledCoursesResponse;
+import org.edx.mobile.model.course.CourseComponent;
+import org.edx.mobile.model.course.DiscussionBlockModel;
+import org.edx.mobile.services.ViewPagerDownloadManager;
+
+public class CourseUnitDiscussionFragment extends CourseUnitFragment {
+    public static final String FRAG_TAG = "discussion_frag";
+
+    /**
+     * Create a new instance of fragment
+     */
+    static CourseUnitDiscussionFragment newInstance(CourseComponent unit, EnrolledCoursesResponse courseData) {
+        CourseUnitDiscussionFragment f = new CourseUnitDiscussionFragment();
+        Bundle args = new Bundle();
+        args.putSerializable(Router.EXTRA_COURSE_UNIT, unit);
+        args.putSerializable(Router.EXTRA_ENROLLMENT_COURSE_DATA, courseData);
+        f.setArguments(args);
+        return f;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    /**
+     * The Fragment's UI is just a FrameLayout which nests the CourseDiscussionPostsThreadFragment.
+     */
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_course_unit_discussion, container, false);
+    }
+
+    @Override
+    public void onViewCreated(View view, Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        // First we need to get the discussion topic id to send to the posts fragment
+        String topicId = ((DiscussionBlockModel)getArguments().getSerializable(
+                Router.EXTRA_COURSE_UNIT)).getData().topicId;
```

Comment:
You could also get this by calling the `getCourseComponent()` method in the `CourseUnitFragment` superclass.